src/NServiceBus.Transport.RabbitMQ/Receiving/MessagePump.cs

```diff
@@ -180,6 +184,7 @@ public async Task StopReceive(CancellationToken cancellationToken = default)
                 {
                     connectionShutdownCompleted.SetResult(true);
                 }
+                connection.Dispose();
```

Comment:
stop should probably also dispose the `messageProcessingCancellationTokenSource` and `messagePumpCancellationTokenSource` which is currently only done in Dispose and would therefore not dispose any intermediary token sources.